base/loading.jl

```diff
@@ -1748,6 +1748,15 @@ function compilecache_path(pkg::PkgId, prefs_hash::UInt64)::String
         crc = _crc32c(something(Base.active_project(), ""))
         crc = _crc32c(unsafe_string(JLOptions().image_file), crc)
         crc = _crc32c(unsafe_string(JLOptions().julia_bin), crc)
+        # TODO: Hash options that have effect codegen
+        # -C: Should not hash on, we support multi-versioning, but we need to choose compatible
+        #     cache file.
+        # --optimize: ??? We might want to cache a higher optimization level than runtime
+        # --min-optlevel: ??? same?
+        # -g ???
+        # --math-mode :sweat:
+        # --inline / --check-bounds
```

Comment:
What happens if packages gets precompiled as part of a `Pkg.test` (where `--check-bounds=yes`). Does that mean that you currently get "stuck" with compiled code with forced bounds checking?

Reply:
Yes on this PR you currently do. My plan is to disable the objcache when certain options are set.